### PR TITLE
Add -value-avro-schema and -value-avro-record-name to produce command

### PR DESCRIPTION
**Pull request by sixstone-qq**
This PR gives support to produce Avro formatted data using AVSC file and
record name to know which schema identifier should be used.

There two new flags have been added:

1. `hkt produce -topic topic-3 -registry http://localhost:8081 -value-avro-schema band.avsc`
To guess the version to used based on the file

2. `hkt produce -topic topic-2 -registry http://localhost:8081 -valuecodec avro -value-avro-record-name r`

To guess the subject name based on TopicRecordNameStrategy, that way, the subject to look for latest version
will be `topic-2-r`.

3. `hkt produce -topic topic-3 -registry http://localhost:8081 -value-avro-schema band.avsc -value-avro-record-name r`

It will use the specific version from the schema file and it will use TopicRecordNameStrategy to find the subject, namely
in this case `topic-3-r` too.

Along with those changes, there is a refactor to have enconding and decoding in a separate file to ease maintainance.

### CODE REVIEW

File: produce.go [hunk 1]
```diff
@@ -97,11 +97,19 @@ func (cmd *produceCmd) run(args []string) error {
 	cmd.partitioner = partitioner
 	var err error
 
+	if cmd.avscFile != "" {
+		// Override -valuecodec flag
+		cmd.valueCodecType = "avro"
+		if err := cmd.loadSchemaFile(cmd.avscFile); err != nil {
```
Comment (tafsuth) [inline produce.go:103]: Here we should pass the de-referenced schema file right ?

**Review by tafsuth — APPROVED** [review]
✅ for the functionality. 

One remark I have is, given that we have the reference now, and there is no straight forward way to get the avsc file (unless running the script to dereference them), it would have been easier to give the user the choice to provide the subject version to use.

File: README.md [hunk 2]
```diff
@@ -237,7 +242,21 @@ $ hkt consume -registry http://localhost:8081 -valuecodec avro actors
...
+  "offset": 1,
+  "key": "id-43",
+  "value": {
+    "FirstName": "Arnold"
+  },
+  "time": "2021-11-17T01:01:01Z"
+}
 ```
+
+Not only `TopicNameStrategy` is supported but `TopicRecordNameStrategy` is supported to find the subject name for
```
Comment (Axelcouty) [inline README.md:256]: :muscle:

**Review by Axelcouty — APPROVED** [review]

File: produce.go (hunk 1 above)
Comment (sixstone-qq) [reply produce.go:103]: yes, we don't handle references :)

**Comment by sixstone-qq** [comment]
> One remark I have is, given that we have the reference now, and there is no straight forward way to get the avsc file (unless running the script to dereference them), it would have been easier to give the user the choice to provide the subject version to use.

That's why the #40 is open to fix this issue :)

**Pull request merged**